index.js

```diff
@@ -90,29 +93,36 @@ exports = module.exports = function (app, settings) {
   }
 
 
-  app.context.render = function *(view, _context) {
-    var context = {};
-    merge(context, this.state);
-    merge(context, _context);
-
-    var html = yield *render(view, context);
-
-    var layout = context.layout === false ? false : (context.layout || settings.layout);
-    if (layout) {
-      // if using layout
-      context.body = html;
-      html = yield *render(layout, context);
-    }
-
-    var writeResp = context.writeResp === false ? false : (context.writeResp || settings.writeResp);
-    if (writeResp) {
-      //normal operation
-      this.type = 'html';
-      this.body = html;
-    }else{
-      //only return the html
-      return html;
-    }
+  app.context.render = async function (view, _context) {
+    const ctx = this;
+    return new Promise(async (resolve, reject) => {
```

Comment:
write with async function is just like write code with co, no need to change the logic.